### ISSUE TITLE
Instrumentation causes a duplicate method and a `ClassFormatError` at runtime

### ISSUE THREAD

**Issue by dmdashenkov** [gradle/gradle#13325]
I've got a plugin written in Java. One of the plugin classes works with an external API which required a serializable lambda (to be precise, a `java.util.function.Consumer<..> & Serializable`). 

When running unit tests with that class, everything goes OK. However, when running an integration test using `GradleRunner`, an exception is thrown:
```
FAILURE: Build failed with an exception.
* What went wrong:
  Duplicate method name "$deserializeLambda$" with signature "(Ljava.lang.invoke.SerializedLambda;)Ljava.lang.Object;" in class file io/spine/tools/gradle/protoc/ProtobufGenerator
* Try:
  Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Exception is:
  java.lang.ClassFormatError: Duplicate method name "$deserializeLambda$" with signature "(Ljava.lang.invoke.SerializedLambda;)Ljava.lang.Object;" in class file io/spine/tools/gradle/protoc/ProtobufGenerator
```
Note that depending on the JRE the error message might look a bit different. The one above is from OpenJDK 8 for Linux. Some other JREs might not give the method name.

The class in `io/spine/tools/gradle/protoc/ProtobufGenerator` is the class which calls the API with lambdas.

When substituting lambdas with nested classes, the problem goes away.

It's worth noting that the class uses more than one distinct serializable lambda.

This is reproducible with Gradle 6.5 and not reproducible with Gradle 6.4.1.

Seems like this has something to do with the [InstrumentingTransformer](https://github.com/gradle/gradle/blob/bce503b75b2511e2a4e1951938b2c64d0f6e913f/subprojects/core/src/main/java/org/gradle/internal/classpath/InstrumentingTransformer.java).

### Expected Behavior

Plugin class is loaded successfully.

### Current Behavior

Loading the plugin class causes a `ClassFormatError`.

### Environment

Build scan URL: https://scans.gradle.com/s/bwbp24yp3jzfk
Plugin sources: https://github.com/SpineEventEngine/bootstrap/tree/kotlin-gradle-scripts

**Comment by dmdashenkov** [comment]
@adammurdoch, sorry for bothering, but could you take a look, please? It seems you've authored the code which causes the failure.

This blocks us from migrating to Gradle 6.5 and beyond.

**Comment by adammurdoch** [comment]
Fixed via https://github.com/gradle/gradle/pull/13917. The fix will be in Gradle 6.6

**Comment by dmdashenkov** [comment]
Thanks, looking forward to the release.
